File: system-test/typescript/src/testkit/dsl/port/then/steps/then-step.ts
import type { ThenOrder } from './then-order.js';
import type { ThenCoupon } from './then-coupon.js';
import type { ThenClock } from './then-clock.js';
import type { ThenProduct } from './then-product.js';
import type { ThenCountry } from './then-country.js';
import type { ThenBrowseCoupons } from './then-browse-coupons.js';

export interface ThenStep<TThen> extends PromiseLike<void> {
  and(): TThen;
  order(): ThenOrder;
  order(orderNumber: string): ThenOrder;
  coupon(): ThenCoupon;
  coupon(couponCode: string): ThenCoupon;
  clock(): ThenClock;
  product(skuAlias: string): ThenProduct;
  country(countryAlias: string): ThenCountry;
  coupons(): ThenBrowseCoupons;
}

File: system/multitier/backend-clean-java/src/testSupport/java/com/mycompany/myshop/backend/testkit/dsl/port/then/steps/base/ThenStep.java
package com.mycompany.myshop.backend.testkit.dsl.port.then.steps.base;

import com.mycompany.myshop.backend.testkit.dsl.port.then.steps.ThenClock;
import com.mycompany.myshop.backend.testkit.dsl.port.then.steps.ThenCoupon;
import com.mycompany.myshop.backend.testkit.dsl.port.then.steps.ThenCountry;
import com.mycompany.myshop.backend.testkit.dsl.port.then.steps.ThenOrder;
import com.mycompany.myshop.backend.testkit.dsl.port.then.steps.ThenOrderHistory;
import com.mycompany.myshop.backend.testkit.dsl.port.then.steps.ThenProduct;

public interface ThenStep<T> {
    T and();

    ThenOrder order();

    ThenOrder order(String orderNumber);

    ThenCoupon coupon();

    ThenCoupon coupon(String couponCode);

    ThenOrderHistory orderHistory();

    ThenProduct product(String sku);

    ThenClock clock();

    ThenCountry country(String code);
}
